test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeArrayTests.java

```diff
@@ -714,40 +725,7 @@ static public void testEmptyValueArrayElement() {
 		compareEmptyValArrays(valArr1, valArr2);
 	}
 
-	static void arrayElementStoreNull(Object[] arr, int index) {
-		arr[index] = null;
-	}
-
 	static void arrayElementStore(Object[] arr, int index, Object obj) {
 		arr[index] = obj;
 	}
-
-	@Test(priority=1,invocationCount=2)
-	static public void testStoreNullToNullRestrictedArrayElement1() throws Throwable {
-		int ARRAY_LENGTH = 10;
-		SomePrimitiveClassWithDoubleField[] dstData = new SomePrimitiveClassWithDoubleField[ARRAY_LENGTH];
-
-		try {
-			arrayElementStoreNull(dstData, ARRAY_LENGTH/2);
-		} catch (NullPointerException npe) {
-			return; /* pass */
-		}
-
-		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
-	}
-
-	@Test(priority=1,invocationCount=2)
-	static public void testStoreNullToNullRestrictedArrayElement2() throws Throwable {
-		int ARRAY_LENGTH = 10;
-		SomePrimitiveClassWithDoubleField[] dstData = new SomePrimitiveClassWithDoubleField[ARRAY_LENGTH];
-		Object obj = null;
-
-		try {
-			arrayElementStore(dstData, ARRAY_LENGTH/2, obj);
-		} catch (NullPointerException npe) {
-			return; /* pass */
-		}
-
-		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
-	}
```

Comment:
May I ask you to add these tests back again and include versions of them in both the lw5 and qtypes versions of these files?  As things stand in the lw5 version, the tests should expect `ArrayStoreException` to be thrown.

Although it might look like they repeat testing that's performed elsewhere, they were actually added because of a bug in the JIT compiler - it was making a transformation that was incorrect if a null reference was stored to an array whose component type was a primitive class if the JIT could determine that the reference that was being stored was definitely a null reference.  In the other tests in this file, the JIT can't easily determine whether a null reference is the value that's being stored.

Reply:
With lw5 when a nullrestricted field is cast to a non nullrestricted field it does not keep its nullrestricted properties. Because of that I'm not sure there is an equivalent behavior to these tests.

Reply:
These tests involve arrays rather than fields.  As described in the [previous comment](#discussion_r1396160132), it looks like an `ArrayStoreException` must be thrown in these cases.

Reply:
Again, if the goal is to have things work the same as lw5 as it is, rather than where the specification seems to be going, these test methods should be carried over to the `src_lw5` directory expecting that `NullPointerException`s will be thrown.

Reply:
In the q type world, there are both `value` keyword and `primitive` keyword that can be used to define classes. The `ACC_PRIMITIVE` flag in the class will cause an NPE if null is store to the type.

However, in lw5 `primitive` keyword is removed and the class itself won't have a flag saying it is null-restricted or not. A new CheckedType mechanism will be added in OpenJDK (not added yet) for the null store checks. Eventually OpenJ9 could need 2 J9Classes (1 reference type and 1 null restirected type) for a value class to link to the 2 Checked Types. But before that I guess in lw5 OpenJ9 won't be able to distinguish between the value class array vs primitive value type array like in the q type world.